met-web/src/components/engagement/view/EmailModal.tsx

```diff
@@ -40,6 +40,10 @@ const EmailModal = ({ defaultPanel, open, handleClose }: EmailModalProps) => {
                 email_address: email,
                 survey_id: savedEngagement.surveys[0].id,
             });
+            window.snowplow('trackSelfDescribingEvent', {
+                schema: 'iglu:ca.bc.gov.met/verify_email/jsonschema/1-0-0',
+                data: { survey_id: savedEngagement.surveys[0].id, engagement_id: savedEngagement.id },
```

Comment:
any chance here surveys[0] is undefined and could cause error due to indexing undefined?

Reply:
unlikely since it would break the above methods too